### DISCUSSION TITLE
Validators should not have to return a value.

### DISCUSSION THREAD

**Discussion by billytetrud** [pydantic/pydantic#4239] [Feature Request]
[The documentation writes](https://pydantic-docs.helpmanual.io/usage/validators/) that:

> validators should either return the parsed value or raise a ValueError, TypeError, or AssertionError (assert statements may be used).

However, all the examples seem to simply return the *passed* value. No *parsing* is being done. It is verbose for every validator to have to `return v`. It would be nice for validators to simply not need to return any value on success. 

It doesn't make sense to me why you would return a parsed value from a validator anyway. If you have a chain of validators (eg root_validators), could they possibly mutate the value each run? Validators shouldn't be mutating values. If any parsing or mutation is needed, it would be best designed as separate annotated methods.

**Comment by samuelcolvin** [comment]
I completely disagree.

There's no such thing in python as "not return a value", you can return `None`, but that's not the same as not returning a value - there's no way for pydantic to know if a function has an explicit return statement.

The ability to change the value in a validator is a key feature of pydantic. See the "Note" [here](https://pydantic-docs.helpmanual.io/usage/models/)

**Reply by billytetrud** [reply]
> There's no such thing in python as "not return a value", you can return None

That is exactly what I meant. I meant to not require an explicit return statement.

> there's no way for pydantic to know if a function has an explicit return statement

Obviously. 

> See the "Note" [here](https://pydantic-docs.helpmanual.io/usage/models/)

Interesting. I wasn't aware that the intention of pydantic is to primarily be a parser. Regardless, what I'm asking for is fundamentally not a huge change to the library. At simplest, it could be an option passed to an annotation like `@root_validator(no_mutation=True)`, which would mean that the return value is expected to be None and the value passed in is used as the return value.

**Reply by samuelcolvin** [reply]
Wouldn't be hard to do, particular in pydantic V2, but I doubt if a significant number of people actually want it.
